src/transactions/simulation/TxSimUtils.cpp

```diff
@@ -109,6 +109,50 @@ generateScaledOfferID(OperationResult const& result, uint32_t partition)
     return generateScaledOfferID(offerID, partition);
 }
 
+Hash
+generateScaledClaimableBalanceID(OperationResult const& result,
+                                 uint32_t partition)
+{
+    if (result.code() != opINNER)
+    {
+        // This will produce txINTERNAL_ERROR on apply
+        throw std::runtime_error(
+            "generateScaledOfferID: invalid OperationResult");
+    }
+
+    if (result.tr().type() != CREATE_CLAIMABLE_BALANCE)
+    {
+        throw std::runtime_error(
+            "Invalid OperationResult: must be CREATE_CLAIMABLE_BALANCE");
+    }
+
+    auto const& balanceID =
+        result.tr().createClaimableBalanceResult().balanceID().v0();
+    return generateScaledClaimableBalanceID(balanceID, partition);
+}
+
+Hash
+generateScaledClaimableBalanceID(Hash const& balanceID, uint32_t partition)
+{
+    if (partition == 0)
+    {
+        return balanceID;
+    }
+
+    auto scaledID = balanceID;
+
+    // XOR balanceID and partition
```

Comment:
agree `SHA256(oldID | partition)` is the better way to do this

Reply:
I didn't hash the partition because I didn't want to affect the performance of simulation. Is this not a concern? I would expect an additional SHA256 call for each scaled account/issuer on every op to be a non-negligible performance hit. 

The chances of having a collision with XOR in this case is extremely low (Jon's response to Marta's question . You might have to paste this link into slack first - https://stellarfoundation.slack.com/archives/C0367MGFL/p1595610545101900?thread_ts=1595465813.064400&cid=C0367MGFL).

If you think we should still go with the hash though, please let me know.

Reply:
ah yes, this is fine. it is a hash already after all